1 Trabajo Base de Datos (30 puntos)
Para responder las preguntas siguientes, utilice la base de datos ceosal1 de la librería wooldridge.
Asegúrese de cargarla correctamente (sin nulos) y de especificar el programa de forma rigurosa para
evitar cambios por aleatoriedad
1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?

In [13]:
install.packages("wooldridge")
library(wooldridge)

set.seed(1)

data("ceosal1")

ceosal1 <- na.omit(ceosal1)

head(ceosal1)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



,salary,pcsalary,sales,roe,pcroe,ros,indus,finance,consprod,utility,lsalary,lsales
,<int>,<int>,<dbl>,<dbl>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>,<dbl>
1,1095,20,27595.0,14.1,106.4,191,1,0,0,0,6.998509,10.225389
2,1001,32,9958.0,10.9,-30.6,13,1,0,0,0,6.908755,9.206132
3,1122,9,6125.9,23.5,-16.3,14,1,0,0,0,7.022868,8.720281
4,578,-9,16246.0,5.9,-25.7,-21,1,0,0,0,6.359574,9.695602
5,1368,7,21783.2,13.8,-3.0,56,1,0,0,0,7.221105,9.988894
6,1145,5,6021.4,20.0,1.0,55,1,0,0,0,7.043160,8.703075


INTERPRETACIÓN

La base ceosal1 contiene información de empresas y sus CEO. La variable que normalmente se busca explicar es el salario (salary), mientras que variables como sales, roe y ros se usan para ver qué factores influyen en ese salario.

Hay variables continuas como salary, sales, roe y ros, ya que toman muchos valores distintos. También hay variables tipo dummy como indus, finance, consprod y utility, que indican el tipo de empresa y solo toman valores 0 o 1.

Este tipo de base se usa principalmente para analizar qué determina el salario de los ejecutivos, considerando el tamaño y desempeño de la empresa.

2. (5 puntos) Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable

In [16]:
prom_sales <- mean(ceosal1$sales)

ceosal1$sales_mayor_promedio <- ifelse(ceosal1$sales > prom_sales, 1, 0)

table(ceosal1$sales_mayor_promedio)


  0   1 
155  54 

### INTERPRETACION

Se observa que 155 empresas están bajo el promedio de ventas y 54 lo superan. Esto muestra que la mayoría tiene ventas más bajas y solo un grupo menor concentra las ventas altas.

Esta variable permite separar la muestra y analizar después si el salario del CEO cambia según el nivel de ventas.

3. (5 puntos) Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?

In [18]:
ceosal1$sales_mayor_promedio_indus <- ceosal1$sales_mayor_promedio * ceosal1$indus

table(ceosal1$sales_mayor_promedio_indus)


  0   1 
188  21 

INTERPRETACION

Esta variable junta dos cosas: que la empresa sea industrial y que además tenga ventas sobre el promedio.

En los resultados se ve que solo 21 empresas cumplen ambas condiciones, mientras que 188 no. O sea, son pocas las empresas industriales que además tienen ventas altas.

Esto sirve para identificar ese grupo más específico y después ver si se comporta distinto al resto, por ejemplo en el salario de los CEO.

4. (5 puntos) Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?

In [19]:
ceosal1$sales_mayor_promedio_roe <- ceosal1$sales_mayor_promedio * ceosal1$roe

summary(ceosal1$sales_mayor_promedio_roe)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   4.251   3.500  43.700 

INTERPRETACION

Esta variable combina el ROE con la condición de que la empresa tenga ventas sobre el promedio. Cuando no se cumple, el valor es 0; cuando sí, toma el valor del ROE.

Se observa que la mediana es 0, lo que indica que muchas empresas no superan el promedio de ventas. En cambio, en las que sí lo superan aparecen valores positivos de ROE, llegando hasta 43.7.

Esto permite analizar el ROE solo en las empresas con mayores niveles de ventas y ver si su comportamiento es distinto.

5. (5 puntos) Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [20]:
ceosal1$indus_o_finance <- ifelse(ceosal1$indus == 1 | ceosal1$finance == 1, 1, 0)

table(ceosal1$indus_o_finance)


  0   1 
 96 113 

INTERPRETACION

La variable indus_o_finance identifica a las empresas que pertenecen al sector industrial o financiero, tomando valor 1 en esos casos y 0 en el resto.

A partir de los resultados, se observa que 113 empresas pertenecen a alguno de estos dos sectores, mientras que 96 no.

Esto permite agrupar ambos tipos de empresas en una sola categoría y compararlas más adelante con el resto de la muestra.

2 Modelos de Regresión Lineal (70 puntos)
1. (15 puntos) Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?

In [21]:
modelo <- lm(salary ~ sales + roe + ros, data = ceosal1)

summary(modelo)


Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


INTERPRETACION

Se estimó un modelo donde el salario del CEO depende de sales, roe y ros.
Los resultados muestran que ninguna de las variables es significativa al 5%, ya que todos sus p-values son mayores a 0.05. En particular, roe está más cerca de ser significativa, pero no alcanza ese nivel.
En cuanto al modelo completo, el p-value del test F es 0.083, por lo que tampoco es significativo al 5%.

En general el modelo no logra explicar de forma significativa el salario del CEO con estas variables.

2. (10 puntos) Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?

In [22]:
modelo2 <- lm(salary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe, data = ceosal1)

summary(modelo2)


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


INTERPRETACION

Al agregar las nuevas variables, se busca ver si las empresas con ventas sobre el promedio tienen un comportamiento distinto y si el efecto del ROE cambia en ese grupo.

Sin embargo, ninguna de estas variables resulta significativa, ya que sus p-values son altos. Esto sugiere que no tienen un efecto claro sobre el salario.

Además, el modelo en su conjunto tampoco es significativo al 5%, por lo que no logra explicar de forma relevante el salario del CEO.

3. (15 puntos) Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%.

In [24]:
F_critico <- qf(0.95, 2, 203)

R2_r <- 0.03191
R2_ur <- 0.04059

F_est <- (R2_ur - R2_r)/(1 - R2_ur) * 203 / 2

F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}

[1] 0.9182935

[1] 3.040379

[1] "No se rechaza H0"

INTERPRETACION

El test F evalúa si las dos variables agregadas son significativas en conjunto.

Como el F estimado es 0.918 y es menor que el F crítico de 3.040, no se rechaza la hipótesis nula al 5%.

Por lo tanto, no hay evidencia suficiente para decir que la variable de ventas sobre el promedio y su interacción con el ROE aporten al modelo.

4. (15 puntos) Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.

In [35]:
ceosal1$sales_D <- ceosal1$sales * ceosal1$sales_mayor_promedio
ceosal1$roe_D <- ceosal1$roe * ceosal1$sales_mayor_promedio
ceosal1$ros_D <- ceosal1$ros * ceosal1$sales_mayor_promedio


modelo_grupos <- lm(salary ~ sales + roe + ros + sales_mayor_promedio + sales_D + roe_D + ros_D,
                    data = ceosal1)

summary(modelo_grupos)


F_critico <- qf(0.95, 4, 201)

R2_ur <- summary(modelo_grupos)$r.squared
R2_r <- summary(modelo)$r.squared
F_est <- (R2_ur - R2_r) / (1 - R2_ur) * 201 / 4
F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

[1] 1.517033

[1] 2.416574

[1] "No se rechaza H0"

INTERPRETACION

Este test busca ver si el modelo del salario cambia entre empresas con ventas altas y bajas.

En este caso, el F estimado es aproximadamente 1.52 y el F crítico 2.42. Como el F estimado es menor, no se rechaza la hipótesis nula al 5%.

En simple, no hay evidencia para decir que el comportamiento del salario sea distinto entre ambos grupos de empresas.

5. (15 puntos) Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.

In [37]:
ceosal1_indus <- subset(ceosal1, indus == 1)
ceosal1_finance <- subset(ceosal1, finance == 1)
ceosal1_consprod <- subset(ceosal1, consprod == 1)
ceosal1_utility <- subset(ceosal1, utility == 1)

modelo_restricto <- lm(salary ~ sales + roe + ros, data = ceosal1)

modelo_indus <- lm(salary ~ sales + roe + ros, data = ceosal1_indus)
modelo_finance <- lm(salary ~ sales + roe + ros, data = ceosal1_finance)
modelo_consprod <- lm(salary ~ sales + roe + ros, data = ceosal1_consprod)
modelo_utility <- lm(salary ~ sales + roe + ros, data = ceosal1_utility)

SSR_r <- sum(residuals(modelo_restricto)^2)

SSR_indus <- sum(residuals(modelo_indus)^2)
SSR_finance <- sum(residuals(modelo_finance)^2)
SSR_consprod <- sum(residuals(modelo_consprod)^2)
SSR_utility <- sum(residuals(modelo_utility)^2)

SSR_ur <- SSR_indus + SSR_finance + SSR_consprod + SSR_utility

G <- 4
k <- 3
n <- nrow(ceosal1)

gl1 <- (G - 1) * (k + 1)
gl2 <- n - G * (k + 1)

F_est <- ((SSR_r - SSR_ur) / gl1) / (SSR_ur / gl2)

F_critico <- qf(0.95, gl1, gl2)

F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}

[1] 1.398349

[1] 1.802614

[1] "No se rechaza H0"

INTERPRETACION

El test de Chow evalúa si el modelo del salario cambia entre sectores.
Como el F estimado ( aproximado a 1.40) es menor que el F crítico ( aproximado a 1.80), no se rechaza la hipótesis nula al 5%.

Por lo tanto, no hay evidencia de que la relación entre salary, sales, roe y ros difiera entre sectores.